src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -103,21 +98,21 @@ private static int GetNestedProxiesCount(HttpContext context, IFunctionExecution
                 return new NotFoundResult();
             }
 
-            if (context.Request.IsColdStart())
+            if (context.Request.IsColdStart() && !context.Items.ContainsKey(ScriptConstants.AzureFunctionsColdStartKey))
             {
```

Comment:
The rest of the items modifying context items was already being removed on reentrancy. here is an example: [https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Proxy/ProxyFunctionExecutor.cs#L67](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Proxy/ProxyFunctionExecutor.cs#L67)

So we are fine for rest.